### DISCUSSION TITLE
Strongswan, FreeRadius and Yubikey OTP Integration

### DISCUSSION THREAD

**Discussion by systemxcom** [strongswan/strongswan#199] [Q&A]
Hi,

If you could please provide us with persons that can assist, through a paid engagement, to accomplish a Strongswan, FreeRadius and Yubikey OTP Integration.  We would like to use the strategy that enables use of native IPSEC clients by concatenating the password with the OTP response from a Yubikey Authentication Server.  We have a complete lab set up with all components installed and operational that enables FreeRadius authenticate Strongswan without Yubikey OTP. 

Thank You

**Comment by tobiasbrunner** [comment]
Please check the list [here](https://wiki.strongswan.org/projects/strongswan/wiki/CommercialSupport). However, it sounds like you primarily need someone who knows FreeRADIUS really well, so you might better ask on a support platform related to that.

**Reply by systemxcom** [reply]
Hi Tobias,

Is it possible for your team to develop the described capability through a paid engagement?

Regards

**Reply by tobiasbrunner** [reply]
Unfortunately, not for the foreseeable future as I won't be available for the next two months and it's unclear yet how such development work can be handled afterwards. Also, I really don't know FreeRADIUS. You basically need someone who can get FreeRADIUS (whether through configuration or a plugin/script) to do the same thing that happens on the client (i.e. combine the plain-text user password with the current OTP for that user before it gets fed as password to whatever EAP method is used - most likely EAP-MSCHAPv2 if you are talking about native clients). Maybe [multiOTP](https://github.com/multiOTP/multiotp) and [this guide](https://wiki.freeradius.org/guide/multiOTP-HOWTO) are something you could look into.

**Reply by systemxcom** [reply]
Thank You Tobias
